Source/ACE/Entity/WorldObject.cs

```diff
@@ -168,11 +168,21 @@ public ObjectDescriptionFlag DescriptionFlags
 
         public WeenieHeaderFlag WeenieFlags
         {
-            get { return (WeenieHeaderFlag)AceObject.WeenieHeaderFlags; }
+            get { return SetWeenieHeaderFlag(); }
             protected internal set { AceObject.WeenieHeaderFlags = (uint)value; }
         }
 
-        public WeenieHeaderFlag2 WeenieFlags2 { get; protected set; }
+        public WeenieHeaderFlag2 WeenieFlags2
+        {
+            get
+            {
+                WeenieHeaderFlag2 flags = SetWeenieHeaderFlag2();
+                if (flags != WeenieHeaderFlag2.None)
+                    IncludesSecondHeader = true;
```

Comment:
PhysicsDescriptionFlag, WeenieHeaderFlag and WeenieHeaderFlag2 should really never be set manually by coding. They should always be set by the presence of data, further, WHF and WHF2 are sorta linked in that they both would be set at the same time AND if WHF2 is not None, it needs to set DescriptionFlag so the client knows to expect that data. 

The changes I've made take it out of the hands of coding and puts it squarely on the data to determine what to do. As I see it, this is the best way forward with these fields.